openstates/oh/bills.py

```diff
@@ -13,43 +13,333 @@
+                name = item["amendnum"] + " " + item["version"]
+            else:
+                name = item["name"] or type_of_document
+            link = first_page+item["link"]+"?format=pdf"
+            bill.add_document(name,link,mimetype="application/pdf")
+
+
+    def get_legislator_ids(self,base_url):
+        legislators = {}
+        for chamber in ["House", "Senate"]:
+            doc = self.get(base_url+"chamber/{chamber}/legislators?per_page=100".format(chamber=chamber))
+            leg_json = doc.json()
+            for leg in leg_json["items"]:
+                legislators[leg["med_id"]] = leg["displayname"]
+
+        return legislators
+
+
+    def get_sponsor_name(self,sponsor):
+        return " ".join([sponsor["firstname"],sponsor["lastname"]])
+
+    def process_vote(self,votes,url,bill,legislators,chamber_dict,vote_results):
+        for v in votes["items"]:
+            try:
+                chamber = chamber_dict[v["chamber"]]
+            except KeyError:
+                chamber = "lower" if "h" in v["assocdoc"] else "upper"
+            try:
+                date = datetime.datetime.strptime(v["date"],"%m/%d/%y")
+            except KeyError:
+                try:
+                    date = datetime.datetime.strptime(v["occurred"],"%m/%d/%y")
+                except KeyError:
+                    self.logger.warning("No date found for vote, skipping")
+                    continue
+            try:
+                motion = v["action"]
+            except KeyError:
+                motion = v["motiontype"]
+            try:
+                result = v["results"].lower()
+            except KeyError:
+                result = v["passed"]
+
+            passed = vote_results[result]
+            committee = None
+            if "committee" in v:
+                vote = Vote(chamber,date,motion,passed,0,0,0,yes_votes=[],no_votes=[],other_votes=[],committee=v["committee"])
+            else:
+                vote = Vote(chamber,date,motion,passed,0,0,0,yes_votes=[],no_votes=[],other_votes=[])
+            #the yea and nay counts are not displayed, but vote totals are
+            #and passage status is.
+            if not "yeas" in v:
+                self.logger.warning("there is no actual vote information, skipping")
+                continue
+            for voter_id in v["yeas"]:
+                vote["yes_votes"].append(legislators[voter_id])
+                vote["yes_count"] += 1
+            for voter_id in v["nays"]:
+                vote["no_votes"].append(legislators[voter_id])
+                vote["no_count"] += 1
+            if "absent" in v:
+                for voter_id in v["absent"]:
+                    vote["other_votes"].append(legislators[voter_id])
+                    vote["other_count"] += 1
+            if "excused" in v:
+                for voter_id in v["excused"]:
+                    vote["other_votes"].append(legislators[voter_id])
+                    vote["other_count"] += 1
+
```

Comment:
So I'm not really sure how to do this. @mileswwatkins, any thoughts? The vote types are on the top level, there's no sub-level for votes. I don't like the idea of checking all keys to see if they point to an array of integers since that seems error-prone, but can't think of any other way to do this.